src/irr/asset/CGLIWriter.cpp

```diff
@@ -166,16 +165,26 @@ namespace irr
 				return width * texelBlockByteSize * height * depth;
 			};
 
+			const auto ROW_TEXEL_OR_BLOCK = 0; // TODO
+			const auto blockDimension = getBlockDimensions(imageInfo.format);
+			const auto sourceTexelOrBlockStride = getTexelOrBlockBytesize(imageInfo.format) * ROW_TEXEL_OR_BLOCK;
+
 			for (auto region = image->getRegions().begin(); region != image->getRegions().end(); ++region)
 			{
 				const auto ptrBeginningOfRegion = data + region->bufferOffset;
 				const auto layerSize = getFullSizeOfLayer(region->imageSubresource.mipLevel);
+
+				// size in texels or blocks - but have to check it out
 				const auto textureGliImgHeight = texture.extent(region->imageSubresource.mipLevel).y;
 				const auto textureGliStride = texture.extent(region->imageSubresource.mipLevel).x;
-				const auto textureGliStrideInBytes = textureGliStride * sizeof(singleChannelByteSize);
+				const auto textureGliStrideInBytes = textureGliStride * singleChannelByteSize;
+				const auto textureGLIDepth = texture.extent(region->imageSubresource.mipLevel).z;
+
 				const auto imgBufferWidth = region->bufferRowLength > 0 ? region->bufferRowLength : region->imageExtent.width;
-				const auto imgBufferWidthInBytes = imgBufferWidth * sizeof(singleChannelByteSize);
+				const auto imgBufferWidthInBytes = imgBufferWidth * singleChannelByteSize;
 				const auto imgBufferHeight = region->bufferImageHeight > 0 ? region->bufferImageHeight : region->imageExtent.height;
+
+				const auto pixelOrBlockCount = textureGliImgHeight * textureGliStride; // it's wrong, I have to make sure I get correct value
```

Comment:
that would actually be:
`(imageStrideInTexels/blockSize.x)*(imageHeightInTexels/blockSize.y)*(imageDepth/blockSize.z)`
it would also include the fractional and padding blocks.

GLI and your input region might have different strides and heights.